### PR TITLE
F/yurik/2.0 primer

### PR DESCRIPTION
**Pull request by yurik42**
BMG patches were applied to 2.0 tag
The code was compiled and "tested" with VS 2019

### CODE REVIEW

File: src/software/SfM/main_SfMInit_ImageListing.cpp [hunk 1]
```diff
@@ -127,6 +127,176 @@ std::pair<bool, Vec3> checkPriorWeightsString
...
+				
+				// Remove extension
+				auto last_period_pos = view_base_filename.find_last_of('.');
+				if (view_base_filename.npos != last_period_pos)
+				{
+					view_base_filename = view_base_filename.substr(0, last_period_pos);
+				}
+
+				// Compare the strings
+				if (0 == stricmp(view_base_filename.c_str(), image_name.c_str()))
```
Comment (yurik42) [inline src/software/SfM/main_SfMInit_ImageListing.cpp]: Non-portable use of stricmp ?

Comment (mikec-bmg) [reply src/software/SfM/main_SfMInit_ImageListing.cpp]: It looks like CPL uses a construct like below for a POSIX or Windows version of caseless compare:

```#  if defined(WIN32)
#    define STRCASECMP(a,b)         (stricmp(a,b))
#    define STRNCASECMP(a,b,n)      (strnicmp(a,b,n))
#  else
/** Alias for strcasecmp() */
#    define STRCASECMP(a,b)         (strcasecmp(a,b))
/** Alias for strncasecmp() */
#    define STRNCASECMP(a,b,n)      (strncasecmp(a,b,n))
#  endif
```
Of course, until we actually build for another platform this doesn't really matter, so probably no reason to change it until then.

Comment (mikec-bmg) [reply src/software/SfM/main_SfMInit_ImageListing.cpp]: I have no idea why the 'Insert Code' made the font so big for some lines of that chunk of code.

**Review by mikec-bmg — APPROVED** [review]

File: src/software/SfM/main_SfMInit_ImageListing.cpp (hunk 1 above)
Comment (yurik42) [reply src/software/SfM/main_SfMInit_ImageListing.cpp]: I added a macro definition
```
#ifdef _MSC_VER
#define strcasecmp(x, y) _stricmp(x, y)
#endif
```

File: src/software/SfM/main_SfMInit_ImageListing.cpp [hunk 2]
```diff
@@ -185,9 +366,12 @@ int main(int argc, char **argv)
       << "\n"
       << "[-P|--use_pose_prior] Use pose prior if GPS EXIF pose is available"
       << "[-W|--prior_weights] \"x;y;z;\" of weights for each dimension of the prior (default: 1.0)\n"
-      << "[-m|--gps_to_xyz_method] XZY Coordinate system:\n"
+      << "[-m|--gps_to_xyz_method] XYZ Coordinate system:\n"
       << "\t 0: ECEF (default)\n"
-      << "\t 1: UTM";
+      << "\t 1: UTM\n"
+      << "[-x|--exif_extended_lookup] Allow partial map on camera model name in EXIF information"
```
Comment (vminor) [inline src/software/SfM/main_SfMInit_ImageListing.cpp]: << "[-x|--exif_extended_lookup] Allow partial map on camera model name in EXIF information\n"

**Review by vminor — APPROVED** [review]

File: src/software/SfM/main_SfMInit_ImageListing.cpp (hunk 2 above)
Comment (yurik42) [reply src/software/SfM/main_SfMInit_ImageListing.cpp]: Are you suggesting to add '\n' at the end?
But the original line (190) did not have trailing '\n'

Comment (vminor) [reply src/software/SfM/main_SfMInit_ImageListing.cpp]: yes, unless the [-x] parameter is part of the [-l]. It looks like we added the 3 new help suggestions, and we need a newline to make them look neat.

**Review by mikec-bmg — APPROVED** [review]

**Pull request merged**